code/crates/starknet/host/src/host/proposal.rs

```diff
@@ -149,6 +152,22 @@ async fn run_build_proposal_task(
         }
     }
 
+    // Vote extensions
+    if !vote_extensions.extensions.is_empty() {
+        let transactions = vote_extensions
+            .extensions
+            .into_iter()
+            .map(|(_, e)| e.message)
+            .map(Transaction::new)
+            .collect();
+
+        let part = ProposalPart::Transactions(Transactions::new(transactions));
```

Comment:
maybe make sure they're in multiple parts if we're over the part size.

Reply:
Good point! Moreover, what happens if `value_size + extensions_size > max_block_size`, do we just drop the remaining extensions or do we include them regardless? cc @cason

Reply:
let's just drop. not much else we can do. Note this is just something for our test app so we can do whatever we want here. In a useful app things would be different but wanted something very simple.

Reply:
Why do we broadcast the extensions in the proposal? It should be enough to include the hashes of them. Nodes are supposed to receive them independently.

Reply:
@ancazamfir I just realized we do not really have a `max_part_size`, but rather `max_txs_per_part`. So we can either:

- Use some heuristic to compute the `max_part_size` based on txs seen so far
- Add a `max_part_size` setting, remove `max_txs_per_part`
- Have one part per extension
- Leave it as is

What do you think?

Reply:
> Why do we broadcast the extensions in the proposal? It should be enough to include the hashes of them. Nodes are supposed to receive them independently.

Yes, I'll get to that in a separate PR. For now we are just piggybacking on proposal parts gossip to disseminate the extensions.

Reply:
> @ancazamfir I just realized we do not really have a `max_part_size`, but rather `max_txs_per_part`. So we can either:
> 
> * Use some heuristic to compute the `max_part_size` based on txs seen so far
> * Add a `max_part_size` setting, remove `max_txs_per_part`
> * Have one part per extension
> * Leave it as is
> 
> What do you think?

Do we have access to the `vote_extensions.size` config? If not we can just count one extension as one tx for now. And maybe in the app when we generate it we use min(`tx_size`, `vote_extensions.size`) since the extension is used as a tx in this app.